Modix.Bot/ModixBot.cs

```diff
@@ -224,14 +231,23 @@ private async Task HandleCommand(SocketMessage messageParam)
 
             var context = new CommandContext(_client, message);
 
-            using (var scope = _scope.ServiceProvider.CreateScope())
-            {
-                await scope.ServiceProvider
-                    .GetRequiredService<IAuthorizationService>()
-                    .OnAuthenticatedAsync(context.User as IGuildUser);
+            var commandScope = _scope.ServiceProvider.CreateScope();
+            _commandScopes[context] = commandScope;
 
-                var result = await _commands.ExecuteAsync(context, argPos, scope.ServiceProvider);
+            await commandScope.ServiceProvider
+                .GetRequiredService<IAuthorizationService>()
+                .OnAuthenticatedAsync(context.User as IGuildUser);
+
+            await _commands.ExecuteAsync(context, argPos, commandScope.ServiceProvider);
+
+            stopwatch.Stop();
+            Log.LogInformation($"Took {stopwatch.ElapsedMilliseconds}ms to process: {message}");
+        }
 
+        private async Task HandleCommandResultAsync(Optional<CommandInfo> command, ICommandContext context, IResult result)
+        {
+            using (var commandScope = _commandScopes[context])
```

Comment:
Would it be appropriate to remove the scope from the collection at this point?

Reply:
Yeah, good idea